.github/workflows/build-&-publish-docker-image.yml

```diff
@@ -47,41 +49,41 @@ jobs:
     - name: Build & Push Miner Docker Image
       run: |
         sed 's,%COPY%,COPY --from=miner_build $APP_DIR,g' "$DOCKERFILE_MINER.template" > "$DOCKERFILE_MINER"
-        docker build --build-arg GIT_COMMIT=$GITHUB_SHA -t $MINER_REGISTRY:$TAG -f "$DOCKERFILE_MINER" .
-        docker tag $MINER_REGISTRY:$TAG $MINER_REGISTRY:latest
-        docker tag $MINER_REGISTRY:$TAG $MINER_REGISTRY:staging
-        docker push $MINER_REGISTRY:$TAG
+        docker build --build-arg GIT_COMMIT=$GITHUB_SHA -t miner:latest -f "$DOCKERFILE_MINER" .
+        
+        docker tag miner:latest ${MINER_REGISTRY}:$TAG
+        docker push ${MINER_REGISTRY}:$TAG
+
+        SHORT_SHA=$(echo $GITHUB_SHA | head -c 8)
+        docker tag miner:latest ${MINER_REGISTRY}:$TAG-$SHORT_SHA
+        docker push ${MINER_REGISTRY}:$TAG-$SHORT_SHA  
+
       env:
-        TAG: ${{ steps.get_info.outputs.IMAGE_TAG }}
         DOCKERFILE_MINER: "docker.local/build.miner/Dockerfile"
 
     - name: Push Miner Docker Image With latest Tag
+      if: ${{ github.event.inputs.latest_tag == 'yes' || github.ref == 'refs/heads/master' }}
```

Comment:
The "latest" tag comes from master, so why not call it master? why is it called "latest" this adds complexity, for example this entire step could be removed

Reply:
this is probably not a conversation for now

Reply:
`latest` is default tag on docker cli